packages/components/src/templates/next/components/internal/Filter/FilterDrawer.tsx

```diff
@@ -0,0 +1,176 @@
+"use client"
+
+import { useEffect, useState } from "react"
+import { Dialog, DialogBackdrop, DialogPanel } from "@headlessui/react"
+import {
+  Button as AriaButton,
+  composeRenderProps,
+  Label,
+} from "react-aria-components"
+import { BiChevronDown, BiX } from "react-icons/bi"
+
+import type { FilterProps } from "../../../types/Filter"
+import { tv } from "~/lib/tv"
+import { focusRing } from "~/utils/focusRing"
+import { Button } from "../Button"
+import { Checkbox, CheckboxGroup } from "../Checkbox"
+import { IconButton } from "../IconButton"
+
+const expandFilterButtonStyle = tv({
+  extend: focusRing,
+  base: "prose-headline-base-semibold flex w-full flex-row items-center justify-between gap-4 text-base-content",
+})
+
+interface FilterDrawerProps extends FilterProps {
+  isOpen: boolean
+  onOpen: (isOpen: boolean) => void
+}
+
+const FilterDrawerContent = ({
+  onOpen,
+  filters,
+  appliedFilters: initialAppliedFilters,
+  handleClearFilter,
+  setAppliedFilters,
+}: FilterDrawerProps) => {
+  const [showFilter, setShowFilter] = useState<Record<string, boolean>>(
+    filters.reduce((acc, { id }) => ({ ...acc, [id]: true }), {}),
+  )
+
+  const [holdingFiltersById, setHoldingFiltersById] = useState(
+    initialAppliedFilters.reduce(
+      (acc, { id, items }) => ({ ...acc, [id]: items.map(({ id }) => id) }),
+      {} as Record<string, string[]>,
+    ),
+  )
+
+  // Synchronize the applied filters with the holding filters
+  useEffect(() => {
+    setHoldingFiltersById(
+      initialAppliedFilters.reduce(
+        (acc, { id, items }) => ({ ...acc, [id]: items.map(({ id }) => id) }),
+        {} as Record<string, string[]>,
+      ),
+    )
+  }, [initialAppliedFilters])
+
+  const updateFilterToggle = (filterId: string) => {
+    setShowFilter((prevFilters) => ({
+      ...prevFilters,
+      [filterId]: !prevFilters[filterId],
+    }))
+  }
+
+  const handleApplyFilters = () => {
+    setAppliedFilters(
+      Object.entries(holdingFiltersById).map(([id, items]) => ({
+        id,
+        items: items.map((id) => ({ id })),
+      })),
+    )
+    onOpen(false)
+  }
```

Comment:
issue: got bug somewhere when unchecking filters instead of using clear all filters

https://github.com/user-attachments/assets/b9a69761-70ae-4dec-88c3-9bd798cd5c6b

Reply:
fixed